### PR TITLE
tune a video: sdv2 and sdv1

### PR DESCRIPTION
**Pull request by wtomin**
The training and inference scripts for Tune-A-Video.
- [x] support training sdv2 (2.1) on 910B(num_frames=24) and 910A (num_frames=12);
- [x] support training sdv1.5 on 910B (num_frames=12) and 910A (num_frames=4);
- [ ] sdv1.5 supports flash-attention in the future

### CODE REVIEW

File: examples/stable_diffusion_v2/ldm/models/diffusion/ddim.py [hunk 1]
```diff
@@ -356,11 +355,15 @@ def encode(
             alphas_next = self.ddim_alphas[:num_steps]
             alphas = self.ddim_alphas_prev[:num_steps]
 
+        timesteps = np.arange(self.ddpm_num_timesteps) if use_original_steps else self.ddim_timesteps
+        timesteps = timesteps[:num_steps]
+        iterator = tqdm(timesteps, desc="Encoding image", total=timesteps.shape[0])
+
         x_next = x0
         intermediates = []
         inter_steps = []
```
Comment (zhtmike) [inline examples/stable_diffusion_v2/ldm/models/diffusion/ddim.py:364]: is `timesteps` and `range(num_steps)` yield the same result in all situations

Comment (wtomin) [reply examples/stable_diffusion_v2/ldm/models/diffusion/ddim.py:364]: No. They are not yielding the same results unless `timesteps` is the same as `range(num_steps)`.

Actually, I think there was a bug before I edited it. I take the decode func ([line 421-432](https://github.com/mindspore-lab/mindone/blob/5d2555df652f0ff7d22644f8a38709f7598182c0/examples/stable_diffusion_v2/ldm/models/diffusion/ddim.py#L421)) as a reference and revised the encode func. The major difference is that decode func flips the timesteps, while the encode func does not.

File: examples/stable_diffusion_v2/train_tuneavideo.py [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+    "2.0": ("sd_v2_base-57526ee4.ckpt", "v2-inference.yaml", 512),
+    "2.0-v": ("sd_v2_768_v-e12e3a9b.ckpt", "v2-vpred-inference.yaml", 768),
+    "1.5": ("sd_v1.5-d0ab7146.ckpt", "v1-inference.yaml", 512),
+    "1.5-wukong": ("wukong-huahua-ms.ckpt", "v1-inference-chinese.yaml", 512),
+}
+_URL_PREFIX = "https://download.mindspore.cn/toolkits/mindone/stable_diffusion"
+_MIN_CKPT_SIZE = 4.0 * 1e9
+
+
+def init_env(args):
```
Comment (hadipash) [inline examples/stable_diffusion_v2/train_tuneavideo.py:47]: replace with `examples.stable_diffusion_v2.common.init_env` for easier refactoring later.

**Review by hadipash — APPROVED** [review]

File: examples/stable_diffusion_v2/train_tuneavideo.py (hunk 2 above)
Comment (wtomin) [reply examples/stable_diffusion_v2/train_tuneavideo.py:47]: I tried to use `examples.stable_diffusion_v2.common.init_env`, but it returns a different `rank_id` compared with the self-defined `init_env`. The former returns the current device id, while the latter always returns `0`.

Using `examples.stable_diffusion_v2.common.init_env`leads to an error in dataloader collection function, which I believe has something to do with the rank id.

Comment (wtomin) [reply examples/stable_diffusion_v2/train_tuneavideo.py]: Fixed.

**Review by zhtmike — APPROVED** [review]

**Review by zhanghuiyao — APPROVED** [review]

File: examples/stable_diffusion_v2/videos/man-skiing.mp4
Comment (SamitHuang) [inline examples/stable_diffusion_v2/videos/man-skiing.mp4:1]: please upload the results using git content url. and remove the videos folder later

**Comment by SamitHuang** [comment]
Not good to couple the video generation code in SD folder

**Comment by SamitHuang** [comment]
> The training and inference scripts for Tune-A-Video.
> 
> * [x]  support training sdv2 (2.1) on 910B(num_frames=24) and 910A (num_frames=12);
> * [x]  support training sdv1.5 on 910B (num_frames=12) and 910A (num_frames=4);
> * [ ]  sdv1.5 supports flash-attention in the future

@wtomin Did you try SDv1.5 + FA after the following PR is merged?
https://github.com/mindspore-lab/mindone/pull/264

**Comment by wtomin** [comment]
> Not good to couple the video generation code in SD folder

Will extract tune-a-video out of stable diffusion v2 in the future.

**Pull request merged**